### ISSUE TITLE
Steps to transcribe in French

### ISSUE THREAD

**Issue by Ca-ressemble-a-du-fake** [m-bain/whisperX#1]
Hi,

Thanks for sharing this work. You wrote that it still needs testing... can I test it in French :wink:? 
I am not sure what I should change. I saw that the wav2vec2 model could be passed in as parameter (see the readme), but in code there are some harcoded [pipelines refering to the the english model](https://github.com/m-bain/whisperX/blob/9f9a00ff4c393d3a4e083b5900705ff200999c1b/whisperx/transcribe.py#L413). For French there is a [wav2vec2 model for French](https://huggingface.co/jonatasgrosman/wav2vec2-large-xlsr-53-french) (never tested since I was relying on Whisper only).

Looking forward to testing this in French!

**Comment by m-bain** [comment]
Hi you can try:
`[VOXPOPULI_ASR_BASE_10K_FR](https://pytorch.org/audio/stable/generated/torchaudio.pipelines.VOXPOPULI_ASR_BASE_10K_FR.html#torchaudio.pipelines.VOXPOPULI_ASR_BASE_10K_FR)`

This wav2vec2.0 model finetuned on french. Let me know how it goes -- I can start setting default models per language

**Comment by Ca-ressemble-a-du-fake** [comment]
Thanks for you reply. I used `whisperx test.wav --model large --output_dir test_whisperx --align_model large --align_extend 2 --language fr` because it did not accept the french model :
`whisperx: error: argument --align_model: invalid choice: 'VOXPOPULI_ASR_BASE_10K_FR' (choose f
rom 'tiny.en', 'tiny', 'base.en', 'base', 'small.en', 'small', 'medium.en', 'medium', 'large-v
1', 'large-v2', 'large')
`
But then it wanted to download `wav2vec2_fairseq_large_lv60k_asr_ls960.pth`.

**Comment by m-bain** [comment]
Ah sorry there was a bug in the argument parser, git pull and try again now :)

**Comment by Ca-ressemble-a-du-fake** [comment]
Ok thanks for the update!
I could test it. If I let the code as is, then there is no improvements over Whisper large model (timestamps are exactly the same with the same inaccuracy). But then if I changed the following line `bundle = torchaudio.pipelines.WAV2VEC2_ASR_LARGE_LV60K_960H` with `bundle = torchaudio.pipelines.VOXPOPULI_ASR_BASE_10K_FR` then it could sometimes find intermediate  timestamps. For example a long sentence starting originally (whisper large) from 26.000 to 35.000  was split into 26.000 to 29.280 and then 29.280 to 35.000. 

Yet I could test further because it then failed during "Alignment" because `tokens = [model_dictionary[c] for c in transcription_cleaned]`. 

I think it has something to do with the apostroph `'`. Moreover accented letters are missing. Here is how transcription_cleaned looks like : `C'EST|UNE|NOUVELLE|DITION` where `DITION` should be `ÉDITION` (note the starting É).

Consequently so far I cannot tell you if it brings improvements in French over stock Whisper:wink:!

**Comment by RaulKite** [comment]
Hi,

I can let a hand with Spanish. I started yesterday with normalization and spanish.py. 

I will continue during next week.

**Comment by m-bain** [comment]
@Ca-ressemble-a-du-fake thanks for pointing this out, I see the culprit was this line
 
` t_words_clean = [re.sub(r"[^a-zA-Z' ]", "", x) for x in t_words]`

Which hard-coded the cleaning of the whisper transcript to only contain a-z and apostrophes (which is in the english wav2vec2.0 dictionary).
I have now changed this cleaning to depend on the alignment model's dictionary in this [commit](https://github.com/m-bain/whisperX/commit/b47bb769a071de8344f46e47c80f1ad11fa1ed4c), so for french this will include accented letters. etc.

@RaulKite keep me updated on spanish etc.

**Comment by Ca-ressemble-a-du-fake** [comment]
Thanks for the update! Why is in converting the text to [upper case](https://github.com/m-bain/whisperX/blob/938341c05a462a257c80be883ac26c497e4bcc0f/whisperx/transcribe.py#L295) ? Because of that it does shows a KeyError with the third character ("E").

If I print the [model_dictionary](https://github.com/m-bain/whisperX/blob/938341c05a462a257c80be883ac26c497e4bcc0f/whisperx/transcribe.py#L296) it shows lower case letters. So lower casing the characters from `transcription_cleaned` made it found all the letters : `tokens = [model_dictionary[c.lower()] for c in transcription_cleaned]` but then it contradicts `upper()` a couple of lines earlier!

Then in utils while writing the transcript to disk in ass format, `prev` can be None (line 194). Consequently I had to add another check : 
`if prev is not None and word['start'] > prev:`.

With these quick and dirty workarounds the output even with large whisper model looks far more accurate!

**Comment by m-bain** [comment]
Thank for this, I have pushed a big fix and it seems working for french now (with varying degrees of accuracy). Please see examples in the README.
Performance on non-english languages could be improved, and I will be investigating this. But closing this for now :)
Feel free to re-open if you find any further issues on FR

**Comment by Ca-ressemble-a-du-fake** [comment]
Thanks for this fix! It works good now! Punctuation is missing but I am not sure this is specific to French, is it ?

**Comment by m-bain** [comment]
@Ca-ressemble-a-du-fake  I think this is a whisper issue. Some examples it does output punctuation: https://github.com/m-bain/whisperX#french

From my experience it seems non-english Whisper tends to transcrbine without subtitles sometimes, sorry

**Comment by RaulKite** [comment]
> Thanks for this fix! It works good now! Punctuation is missing but I am not sure this is specific to French, is it ?

It usually is fixed giving and initial prompt containing punctuation signs. For example: "and now, we continue with next video."

**Comment by Ca-ressemble-a-du-fake** [comment]
@m-bain I only tried with medium and the examples use the large model (which shows much worse timestamps). By the way whisper alone does output punctuation (all models). 

Furthermore I noticed that starting timestamps are often in the middle of the starting words. Is there a way that it starts earlier ? Moreover would you mind explaining very briefly what the wav2vec2 model is used for ? I mean what your agorithm does (I can read it but I still don't understand it)?

Thanks @RaulKite I'll try with initial-prompt also!

**Comment by Ca-ressemble-a-du-fake** [comment]
Awesome! Thanks a ton @m-bain I can look at your code again and learn how you did that:smile: !
